File: python/226-invert-binary-tree.py
# Definition for a binary tree node.
class TreeNode:
  def __init__(self, x):
    self.val = x
    self.left = None
    self.right = None
  def show(self):
    print self.val
    if self.left is not None:
      self.left.show()
    if self.right is not None:
      self.right.show()
    

class Solution:
  # @param {TreeNode} root
  # @return {TreeNode}
  def invertTree(self, root):
    if root is None:
      return root
    self.invertTree(root.left);
    self.invertTree(root.right);
    root.left,root.right = root.right,root.left
    return root

t=TreeNode(1)
s = Solution()
s.invertTree(t);
t.show()

File: js/226-invert-binary-tree.js
var invertTree = function(root) {
  if(root == null){
    return null;
  }
  invertTree(root.left);
  invertTree(root.right);
  root.left = [root.right, root.right=root.left][0];
  return root;
}

function TreeNode(val) {
  this.val = val;
  this.left = this.right = null;
}

function show(root) {
  if(root == null){
    return;
  }
  console.log(root['val']);
  show(root.left);
  show(root.right);
}
t = new TreeNode(1);
t1 = new  TreeNode(2);
t2 = new TreeNode(3);
t.left = t1;
t.right = t2;
show(t);
invertTree(t);
show(t);
